### ISSUE TITLE
Steam cache generic multiple dns?

### ISSUE THREAD

**Issue by dsozzi** [steamcache/steamcache-dns#30]
hi,

I'm going to configure a steamcache next week and I'm getting ready studying all the dockers needed. I'm goinng to support multiple services (steam, origin and battlenet) so I'm going to setup the following cache services:

**STEAM CACHE IP: 10.0.0.1**

````
docker run \
  --restart unless-stopped \
  --name cache-steam \
  -v /cache/steam/data:/data/cache \
  -v /cache/steam/logs:/data/logs \
  -p 10.0.0.1:80:80 \
  steamcache/generic:latest
````

**ORIGIN CACHE IP: 10.0.0.2**
````
docker run \
  --restart unless-stopped \
  --name cache-steam \
  -v /cache/origin/data:/data/cache \
  -v /cache/origin/logs:/data/logs \
  -p 10.0.0.2:80:80 \
  steamcache/generic:latest
````

**BATTLENET CACHE IP: 10.0.0.3**

````
docker run \
  --restart unless-stopped \
  --name cache-steam \
  -v /cache/battlenet/data:/data/cache \
  -v /cache/battlenet/logs:/data/logs \
  -p 10.0.0.3:80:80 \
  steamcache/generic:latest
````

**The first question is, Do I have to run a steamcache-dsn for each of these services right?**

Then, the documentations says:

> The image needs to know the IP of the steamcache server, provide this via the STEAMCACHE_IP variable. If you're also using the generic cache server, you should use USE_GENERIC_CACHE=true and set LANCACHE_IP variable.

My second question is: 
**What should be the LANCACHE_IP? This is the thing that I can't get my head around.**

Given the above configuration and setting up a steamcache-dns for Origin for example, what should be the LANCACHE_IP?

````
docker run --name steamcache-dns -p 53:53/udp -e ORIGINCACHE_IP=10.0.0.2 -e USE_GENERIC_CACHE=true -e LANCACHE_IP=??? steamcache/steamcache-dns:latest
````

Thanks

**Comment by JasonRivers** [comment]
You just need a single DNS image, The document is slightly wrong (I'll look at updating it when I'm slightly less busy than I am right now). It should read "if you're using only a single generic cache server, then you should use USE_GENERIC_CACHE.
You will want to set this value to FALSE and define each cache IP seperately how you have with the ORIGINCACHE_IP - so you'll want:

```
docker run --name steamcache-dns -p 53:53/udp -e ORIGINCACHE_IP=10.0.0.2 -e  STEAMCACHE_IP=10.0.0.1 -e BLIZZARDCACHE_IP=10.0.0.3 steamcache/steamcache-dns:latest
```

Future versions of this image mey remove the USE_GENERIC_CACHE as we have seen cache clashes between different services that run through the same cache container, we no longer recommend using a single cache image for multiple services, so the setup that you have would be correct.

**Comment by dsozzi** [comment]
So, no use of LANCACHE_IP variable? And what about the DNS to configure in my router? Which IP should I enter? Should I define a specific IP for the dns-cache such as
````
docker run --name steamcache-dns -p 10.0.0.99:53:53/udp -e ORIGINCACHE_IP=10.0.0.2 -e  STEAMCACHE_IP=10.0.0.1 -e BLIZZARDCACHE_IP=10.0.0.3 steamcache/steamcache-dns:latest
````
And set into my router thee 10.0.0.99 IP?

thanks again

**Comment by JasonRivers** [comment]
Yep, LANCACHE_IP is one IP for everything, so you wont want that one.
You should point your routers DNS to the steamcache_dns image (or configure DHCP to issue it).

**Comment by dsozzi** [comment]
Just for confirmation, my last snippet was then correct?

````
docker run --name steamcache-dns -p 10.0.0.99:53:53/udp -e ORIGINCACHE_IP=10.0.0.2 -e  STEAMCACHE_IP=10.0.0.1 -e BLIZZARDCACHE_IP=10.0.0.3 steamcache/steamcache-dns:latest
````
the steamcache_dns image IP would be 10.0.0.99 right?
